pkg/auth/authmap_cache.go

```diff
@@ -12,20 +12,24 @@ import (
 	"golang.org/x/exp/maps"
 
 	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/maps/authmap"
+	"github.com/cilium/cilium/pkg/metrics"
 )
 
 type authMapCache struct {
 	logger            logrus.FieldLogger
 	authmap           authMap
 	cacheEntries      map[authKey]authInfo
 	cacheEntriesMutex lock.RWMutex
+	pressureGauge     *metrics.GaugeWithThreshold
```

Comment:
@sayboras is there a specific reason why the pressure gauge needs to live in the cache? why not in the authmap itself? only because we have the number of cached entries for pressure calculation? IMO wrong responsibility :confused:

Reply:
ah I can't find a good way to get the current estimated number of entries in authmap without interating to all entries (which is a costly operation).

Reply:
thought so - and i kind of understand. it's just that IMO this should be the responsibility of the auth map itself. even if that means that we somehow need to keep track of the entries via a different approach - e.g. keeping a counter in the auth map :thinking: 

it's just that the sole purpose of these separate components is to separate their concerns.

Reply:
Agree with Macro's responsibility concerns, just want to add a point that `authMapCache` to some extent can be seen as the facade of authMap, instead of a mere cache. For instance, we must acquire the mutex `cacheEntriesMutex` before accessing `authMap`, and `authMap` cannot be used outside of the `authMapCache`. From this point of view, it might be acceptable to put `pressureGauge` in the `authMapCache`.

Reply:
@jschwinger233 , yes it's a write&read-through cache - which acts as kind of a facade. the main purpose is to immediately reflect the changes in the BPF  map after successful authentication (without a delay to await a sync). 

But there are other "facades" too - like the authmap writer - which translates the userspace representation of the auth map to the bpf auth map representation.

Therefore my concerns whether the cache is the right place :)